plugins/modules/swim_workflow_manager.py

```diff
@@ -2089,38 +2162,40 @@ def verify_diff_tagged(self):
         tag_image_golden = tagging_details.get("tagging")
         image_id = self.have.get("tagging_image_id")
         image_name = self.get_image_name_from_id(image_id)
+        device_role = tagging_details.get("device_role", "ALL")
 
-        image_params = dict(
-            image_id=self.have.get("tagging_image_id"),
-            site_id=self.have.get("site_id"),
-            device_family_identifier=self.have.get("device_family_identifier"),
-            device_role=tagging_details.get("device_role", "ALL").upper()
-        )
-        self.log("Parameters for checking the status of image: {0}".format(str(image_params)), "INFO")
+        for role in device_role.split(','):
+            image_params = dict(
+                image_id=self.have.get("tagging_image_id"),
+                site_id=self.have.get("site_id"),
+                device_family_identifier=self.have.get("device_family_identifier"),
+                device_role=role.upper()
+            )
+            self.log("Parameters for checking the status of image: {0}".format(str(image_params)), "INFO")
 
-        response = self.dnac._exec(
-            family="software_image_management_swim",
-            function='get_golden_tag_status_of_an_image',
-            op_modifies=True,
-            params=image_params
-        )
-        self.log("Received API response from 'get_golden_tag_status_of_an_image': {0}".format(str(response)), "DEBUG")
-
-        response = response.get('response')
-        if response:
-            image_status = response['taggedGolden']
-            if image_status == tag_image_golden:
-                if tag_image_golden:
-                    self.msg = """The requested image '{0}' has been tagged as golden in the Cisco Catalyst Center and
-                             its status has been successfully verified.""".format(image_name)
-                    self.log(self.msg, "INFO")
-                else:
-                    self.msg = """The requested image '{0}' has been un-tagged as golden in the Cisco Catalyst Center and
-                            image status has been verified.""".format(image_name)
-                    self.log(self.msg, "INFO")
-        else:
-            self.log("""Mismatch between the playbook input for tagging/un-tagging image as golden and the Cisco Catalyst Center indicates that
-                        the tagging/un-tagging task was not executed successfully.""", "INFO")
+            response = self.dnac._exec(
+                family="software_image_management_swim",
+                function='get_golden_tag_status_of_an_image',
+                op_modifies=True,
+                params=image_params
+            )
+            self.log("Received API response from 'get_golden_tag_status_of_an_image': {0}".format(str(response)), "DEBUG")
+
+            response = response.get('response')
+            if response:
+                image_status = response['taggedGolden']
```

Comment:
self.log("Current golden tag status for image '{0}': {1}".format(image_name, image_status), "DEBUG")